addon/utils/ref.js

```diff
@@ -17,6 +17,16 @@ export function setGlobalRef(value) {
   lastGlobalRef = value;
 }
 
+export function maybeCleanGlobalRef(value) {
+  if (lastGlobalRef === value) {
+    cleanGlobalRef();
+  }
+}
```

Comment:
I'm not 100% I understand `lastGlobalRef` but I think we should remove it on destruction of a modifier only if it references to that instance.

Reply:
in global ref we always have same ref (if we have only one ember app)

Reply:
@lifeart Hmm, I'm confused a bit. Could you tell me then what `lastGlobalRef` does? 😅 Or suggest how that clean-up should look like?

If `lastGlobalRef` is always pointing to the same thing then I should rather keep only this function (`if` doesn't matter):

```js
export function cleanGlobalRef() {
  lastGlobalRef = null;
}
```

But that would also imply that calling it on the destruction of a modifier isn't the right thing to do, because it will clean it for other "refs" that might rely on that.

Reply:
Okay, I see. So I believe a more proper solution would be to do it through an app initializer and register a deconstructor there. When the app instance is killed, the reference should be nullified.